### PR TITLE
Simple producer benchmark

### PR DESCRIPTION
**Pull request by svroonland**
It's not much, but it gives us something to compare against.

Relates to #1304

### CODE REVIEW

File: zio-kafka-bench/src/main/scala/zio/kafka/bench/ProducerBenchmark.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.util.concurrent.TimeUnit
+
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+class ProducerBenchmark extends ZioBenchmark[Kafka with Producer] {
+  val topic1                      = "topic1"
+  val nrPartitions                = 6
+  val nrMessages                  = 50000
+  val kvs: List[(String, String)] = List.tabulate(nrMessages)(i => (s"key$i", s"msg$i"))
```
Comment (erikvanoosten) [inline zio-kafka-bench/src/main/scala/zio/kafka/bench/ProducerBenchmark.scala:20]: Nice, small messages makes us able to test worst case. Still, 50k small messages is not a lot of traffic yet. I propose we increase that to at least 500k.

File: zio-kafka-bench/src/main/scala/zio/kafka/bench/ProducerBenchmark.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  } yield ()
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  def produceChunk(): Any = runZIO {
+    for {
+      _ <- Producer
+             .produceChunk[Any, String, String](
+               Chunk.fromIterable(kvs.map { case (k, v) =>
+                 new ProducerRecord(topic1, k, v)
```
Comment (erikvanoosten) [inline zio-kafka-bench/src/main/scala/zio/kafka/bench/ProducerBenchmark.scala:37]: Can we move the construction of `ProducerRecord`s outside the benchmark?

**Review by erikvanoosten — COMMENTED** [review]
Great start. I propose we:
* move construction of ProducerRecord to outside the benchmark
* in the benchmark we publish 100 chunks of 5000 small messages

WDYT?

**Review by erikvanoosten — APPROVED** [review]
I just realized that there is one super-tiny thing we can improve: use a binary key/value. Because we now using string, there is the tiny overhead of converting string to byte array. I am not sure if you want to put in the effort, and this is most probably already good.

**Comment by erikvanoosten** [comment]
FYI, I have removed the first run of this new bench from the charts at https://zio.github.io/zio-kafka/dev/bench/ because it was producing a lot less data and is therefore not comparable with later runs.

**Comment by svroonland** [comment]
> I just realized that there is one super-tiny thing we can improve: use a binary key/value. Because we now using string, there is the tiny overhead of converting string to byte array. I am not sure if you want to put in the effort, and this is most probably already good.


I think it doesn't really matter since this isn't currently a benchmark about how much performance we can squeeze out of the Producer. It's just to have a reference for future changes.

**Comment by erikvanoosten** [comment]
Your right, good point 👍

**Comment by svroonland** [comment]
Would be nice to have a "max performance" benchmark though, using the *async methods, batching and clever pipelining.

**Pull request merged**